messenger.rst

```diff
@@ -615,18 +553,166 @@ This middleware can be configured to define the entity manager to use:
+                transports:
+                    async_priority_high:
+                        dsn: '%env(MESSENGER_TRANSPORT_DSN)%'
+
+                        # default configuration
+                        retry_strategy:
+                            max_retries: 3
+                            # milliseconds delay
+                            delay: 1000
+                            # causes the delay to be higher before each retry
+                            # e.g. 1 second delay, 2 seconds, 4 seconds
+                            multiplier: 2
+                            max_delay: 0
+                            # override all of this with a service that
+                            # implements Symfony\Component\Messenger\Retry\RetryStrategyInterface
+                            # service: null
+
+Avoiding Retrying
+~~~~~~~~~~~~~~~~~
+
+Sometimes handling a message might fail in a way that you *know* is permanent
+and should not be retried. If you throw
+:class:`Symfony\\Component\\Messenger\\Exception\\UnrecoverableMessageHandlingException`,
+the message will not be retried.
+
+.. _messenger-failure-transport:
+
+Saving & Retrying Failed Messages
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If a message fails it is retried multiple times (``max_retries``) and then will
+be discarded. To avoid this happening, you can instead configure a ``failure_transport``:
 
-    .. code-block:: xml
+.. configuration-block::
 
-        <!-- config/packages/messenger.xml -->
-        <?xml version="1.0" encoding="UTF-8" ?>
+    .. code-block:: yaml
+
+        # config/packages/messenger.yaml
+        framework:
+            messenger:
+                # after retrying, messages will be sent to the "failed" transport
+                failure_transport: failed
+
+                transports:
+                    # ... other transports
+
+                    failed: 'doctrine://default?queue_name=failed'
+
+In this example, if handling a message fails 3 times (default ``max_retries``),
+it will then be sent to the ``failed`` transport. While you *can* use
+``messenger:consume failed`` to consume this like a normal transport, you'll
+usually want to manually view the messages in the failure transport and choose
+to retry them:
+
+.. code-block:: terminal
+
+    # see all messages in the failure transport
+    $ php bin/console messenger:failed:show
+
+    # see details about a specific failure
+    $ php bin/console messenger:failed:show 20 -vv
+
+    # view and retry messages one-by-one
+    $ php bin/console messenger:failed:retry -vv
+
+    # retry specific messages
+    $ php bin/console messenger:failed:retry 20 30 --force
+
+    # remove a message without retrying it
+    $ php bin/console messenger:failed:retry 20
+
+.. _messenger-transports-config:
+
+Transport Configuration
+-----------------------
+
+Messenger supports a number of different transport types, each with their own
+options.
+
+amqp Transport
```

Comment:
Could we use "the human name" for this title -> `AMQP Transport` instead of the "exact config name" -> `amqp Transport` ?  Currently it reads a bit weird. Same for the other transports, such as Doctrine.